modules/openxr/extensions/openxr_extension_wrapper_extension.cpp

```diff
@@ -0,0 +1,206 @@
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "openxr_extension_wrapper_extension.h"
+#include "../openxr_api.h"
+
+void OpenXRExtensionWrapperExtension::_bind_methods() {
+	GDVIRTUAL_BIND(_get_request_extensions);
+	GDVIRTUAL_BIND(_set_system_properties_and_get_next_pointer, "next_pointer");
+	GDVIRTUAL_BIND(_set_instance_create_info_and_get_next_pointer, "next_pointer");
+	GDVIRTUAL_BIND(_set_session_create_and_get_next_pointer, "next_pointer");
+	GDVIRTUAL_BIND(_set_swapchain_create_info_and_get_next_pointer, "next_pointer");
+	GDVIRTUAL_BIND(_on_register_metadata);
+	GDVIRTUAL_BIND(_on_before_instance_created);
+	GDVIRTUAL_BIND(_on_instance_created, "instance");
+	GDVIRTUAL_BIND(_on_instance_destroyed);
+	GDVIRTUAL_BIND(_on_session_created, "session");
+	GDVIRTUAL_BIND(_on_process);
+	GDVIRTUAL_BIND(_on_pre_render);
+	GDVIRTUAL_BIND(_on_session_destroyed);
+	GDVIRTUAL_BIND(_on_state_idle);
+	GDVIRTUAL_BIND(_on_state_ready);
+	GDVIRTUAL_BIND(_on_state_synchronized);
+	GDVIRTUAL_BIND(_on_state_visible);
+	GDVIRTUAL_BIND(_on_state_focused);
+	GDVIRTUAL_BIND(_on_state_stopping);
+	GDVIRTUAL_BIND(_on_state_loss_pending);
+	GDVIRTUAL_BIND(_on_state_exiting);
+	GDVIRTUAL_BIND(_on_event_polled, "event");
+
+	ClassDB::bind_method(D_METHOD("get_openxr_api"), &OpenXRExtensionWrapperExtension::get_openxr_api);
+	ClassDB::bind_method(D_METHOD("register_extension_wrapper"), &OpenXRExtensionWrapperExtension::register_extension_wrapper);
+}
+
+HashMap<String, bool *> OpenXRExtensionWrapperExtension::get_request_extensions() {
+	Dictionary request_extension;
+
+	if (GDVIRTUAL_CALL(_get_request_extensions, request_extension)) {
```

Comment:
This is more a reminder for later, possibly 4.1, but in hindsight I think parsing a pointer to a bool around isn't really suitable for GDExtensions. We did it this way in the plugin in Godot 3 so I kept that approach, alas.

We might change this to work similar to what we've done on the Vulkan side where we keep everything name based on the extension. So we simply record we want an extension enabled, and whether it is a mandatory extension, and have a separate call that checks whether and extension was enabled.

But I think that is too late now to change so close to 4.0 release.